### PR TITLE
Increase code coverage of Get-ChildItem on file system.

### PR DESCRIPTION
**Pull request by jeffbi**
Addresses part of #4148.

Using the `-Name` parameter on `Get-ChildItem` exercises a part of the file system provider code not previously covered by tests.

### CODE REVIEW

File: test/powershell/Modules/Microsoft.PowerShell.Management/FileSystem.Tests.ps1 [hunk 1]
```diff
@@ -84,6 +84,11 @@ Describe "Basic FileSystem Provider Tests" -Tags "CI" {
             $dirContents.Count | Should Be 2
         }
 
+        It "Verity Get-ChildItem can get the name of a specified item." {
+            $fileName = Get-ChildItem $testFile -Name
+            $fileName | Should BeExactly $fileName
```
Comment (dantraMSFT) [inline test/powershell/Modules/Microsoft.PowerShell.Management/FileSystem.Tests.ps1]: This compares to itself.
I suggest calling Get-ChildItem twice, once with -Name to get the name string and once without to get the fileinfo. The test would look something like the following:
$filename | Should BeExactly $fileInfo.Name

**Review by dantraMSFT — CHANGES_REQUESTED** [review]

File: test/powershell/Modules/Microsoft.PowerShell.Management/FileSystem.Tests.ps1 (hunk 1 above)
Comment (jeffbi) [inline test/powershell/Modules/Microsoft.PowerShell.Management/FileSystem.Tests.ps1]: Yikes! Bad typo on my part. That should have been `$filename | Should BeExactly $testFile`. Is making that change sufficient, or would you prefer the dual-call method?

**Comment by dantraMSFT** [comment]
You're better off comparing the returned value against the name property since, I believe, that's the intent.

**Comment by jeffbi** [comment]
Well, the intent was to check that I got the item I asked for, which is why I was comparing against `$testFile`. But I'll go ahead with your recommendation.

**Review by dantraMSFT — APPROVED** [review]

**Comment by dantraMSFT** [comment]
Sorry jeffbi; I should have been clearer.  When I said the intent of -Name, I mean the intent is to return the Name property of the FileInfo object.
The current implementation of FileInfo 'does' preserve the file name as it was passed in; however, since it's not documented explicitly, I avoid depending on it.
Additionally, I would expect the test to succeed with other values, such as .\filename.txt, ./filename.txt, and /etc/filename.txt
I hope that clarifies my feedback.

**Pull request merged**